### PR TITLE
New keyword: dns.opcode to match DNS opcode - v1

### PR DESCRIPTION
**Pull request by jasonish**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3165

Suricata-Verify tests:
https://github.com/OISF/suricata-verify/pull/127

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):
- PR jasonish-pcap: https://buildbot.openinfosecfoundation.org/builders/jasonish-pcap/builds/373
- PR jasonish: https://buildbot.openinfosecfoundation.org/builders/jasonish/builds/729

### CODE REVIEW

File: src/detect-dns-opcode.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+{
+    SCEnter();
+
+    if (DetectSignatureSetAppProto(s, ALPROTO_DNS) != 0) {
+        return -1;
+    }
+
+    void *detect = rs_detect_dns_opcode_parse(str);
+    if (detect == NULL) {
+        SCLogNotice("Failed to parse dns.opcode: %s", str);
```
Comment (victorjulien) [inline src/detect-dns-opcode.c]: warning/error

Comment (victorjulien) [reply src/detect-dns-opcode.c]: Fixed this up myself.

Comment (jasonish) [reply src/detect-dns-opcode.c]: Oops. Thanks.

**Comment by victorjulien** [comment]
Thanks Jason!

**Pull request merged**